UIforETW/WorkingSet.cpp

```diff
@@ -48,8 +48,13 @@ void CWorkingSetMonitor::SampleWorkingSets()
 	// Allocate enough space to get the working set of most processes.
 	// It will grow if needed.
 	ULONG_PTR numEntries = 100000;
-	std::vector<char> buffer(sizeof(PSAPI_WORKING_SET_INFORMATION) + numEntries * sizeof(PSAPI_WORKING_SET_BLOCK));
-	PSAPI_WORKING_SET_INFORMATION* pwsBuffer = reinterpret_cast<PSAPI_WORKING_SET_INFORMATION*>(&buffer[0]);
+
+	const rsize_t bufferSizeNeeded =
+		sizeof(PSAPI_WORKING_SET_INFORMATION) +
+		(numEntries * sizeof(PSAPI_WORKING_SET_BLOCK));
+
+	std::vector<char> buffer(bufferSizeNeeded);
+	PSAPI_WORKING_SET_INFORMATION* pwsBuffer = reinterpret_cast<PSAPI_WORKING_SET_INFORMATION*>(buffer.data());

```

Comment:
buffer.data() is not clearly better than &buffer[0]. I've always preferred &buffer[0] because it seems very clear and because it works with arrays and vector.